### ISSUE TITLE
[NFR] Ability to suppress progress with --no-progress flag.

### ISSUE THREAD

**Issue by virgofx** [yarnpkg/yarn#1340]
Feature request to add the `--no-progress` flag.

**Why?**
For systems that pipe the stream ... e.g. build systems, CI systems, they don't have ability to render progress via the `\r` modifier or other replacement modifier. 

Composer, NPM and other package manager all include the --no-progress option to suppress progress.

Currently:
![image](https://cloud.githubusercontent.com/assets/739719/19599649/a77accb8-9756-11e6-957d-963cf9d6efa1.png)

Would like:
`yarn install --no-progress`

```
yarn install
v0.16.0
[1/4] Resolving packages...
is-extendable@^0.1.0⠄ is-extendable@^0.1.0⡀ is-extendable@^0.1.0⢀ is-extendable@^0.1.0⠠ is-extendable@^0.1.0⠐ is-extendable@^0.1.0⠈ is-extendable@^0.1.0⠁ is-extendable@^0.1.0⠂ is-extendable@^0.1.0
[2/4] Fetching packages...
warning fsevents@1.0.14: The platform "linux" is incompatible with this module.
info "fsevents@1.0.14" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning Unmet peer dependency "eslint@2.x - 3.x".
warning Unmet peer dependency "eslint@^2.0.0 || ^3.0.0".
[4/4] Building fresh packages...
success Saved lockfile.
Done in 11.65s.
```

**Comment by victornoel** [comment]
See also #788 and #728

By the way, on gitlab-ci, the colors and progress is not present (thanks to #728 which is partially solved).

**Comment by virgofx** [comment]
Thanks @victornoel . Definitely related. I would argue that it would be beneficial to have both options:

`silent` - Does not output STDOUT (ERR only)
`no-progress` - Outputs both STDOUT and STDERR, just no TTY, \R modifiers, Progress information

**Comment by kittens** [comment]
Resolved via #1190.

**Comment by virgofx** [comment]
Thanks ! 👍
